core/src/main/java/com/gentics/mesh/plugin/manager/MeshPluginManagerImpl.java

```diff
@@ -149,20 +221,69 @@ public void startPlugins() {
 					plugin.start();
 					// Set state for PF4J
 					pluginWrapper.setPluginState(PluginState.STARTED);
-					if (plugin instanceof MeshPlugin) {
-						// Set status for Mesh
-						MeshPlugin meshPlugin = (MeshPlugin) plugin;
-						setStatus(meshPlugin.id(), PluginStatus.STARTED);
-						pluginRegistry.preRegister(meshPlugin);
-					}
-					startedPlugins.add(pluginWrapper);
 
-					firePluginStateEvent(new PluginStateEvent(this, pluginWrapper, pluginState));
+					registerMeshPlugin(pluginWrapper.getPluginId());
 				} catch (Throwable e) {
 					log.error("Error while starting plugins " + e.getMessage(), e);
 				}
 			}
 		}
+		if (pluginStorage == restartingPlugins) {
+			pluginStorage.clear();
+		}
+	}
+
+	/**
+	 * Start the specified plugin and its dependencies.
+	 */
+	@Override
+	public PluginState startPlugin(String pluginId) {
+		if (!plugins.containsKey(pluginId)) {
+			throw new IllegalArgumentException(String.format("Unknown pluginId %s", pluginId));
+		}
+
+		PluginWrapper pluginWrapper = getPlugin(pluginId);
+		PluginDescriptor pluginDescriptor = pluginWrapper.getDescriptor();
+		PluginState pluginState = pluginWrapper.getPluginState();
+		if (PluginState.STARTED == pluginState) {
+			log.debug("Already started plugin '{}'", getPluginLabel(pluginDescriptor));
+			return PluginState.STARTED;
+		}
+
+		if (!resolvedPlugins.contains(pluginWrapper)) {
+			log.warn("Cannot start an unresolved plugin '{}'", getPluginLabel(pluginDescriptor));
+			return pluginState;
+		}
+
+		// TODO Are we allowed to enable plugins by ourselves?
```

Comment:
What is the alternative?
Why should a plugin not be enabled once we reach this point?

Reply:
This is the question. In the PF4J's code itself there are cases when the DISABLED status is set during the startup (i.e. if the plugin fails the validation). The plugin may be in theory disabled externally from the config.